Remove support for simple objects in endpoints (#9181)
withastro/astro: .changeset/clever-beds-notice.md

---
'astro': major
---

Removes support for returning simple objects from endpoints (deprecated since Astro 3.0). You should return a `Response` instead.

`ResponseWithEncoding` is also removed. You can refactor the code to return a response with an array buffer instead, which is encoding agnostic.

> The types for middlewares are also cleaned up. To type a middleware function, you should now use `MiddlewareHandler` instead of `MiddlewareResponseHandler`. If you used `defineMiddleware()` to type the function, no changes are needed.

Reviewer: ```suggestion
The types for middlewares have also been revised. To type a middleware function, you should now use `MiddlewareHandler` instead of `MiddlewareResponseHandler`. If you used `defineMiddleware()` to type the function, no changes are needed.
```
Just a suggestion to imply "have changed" vs "cleaned up" since that might not be obvious what that means. So anything like "types for middleware have changed" "have been revised" etc.
